From aba399f4b4d4a475966226831ed2d4b289f9e0eb Mon Sep 17 00:00:00 2001
From: Lewis Liu <wei.zhi.liu@intel.com>
Date: Tue, 4 Mar 2025 00:10:22 +0800
Subject: [PATCH 05/14] fix installation dependent issues. (#6)

* fix installation dependent issues.

Signed-off-by: Liu, Wei Zhi <wei.zhi.liu@intel.com>

* update depedency list.

Signed-off-by: Liu, Wei Zhi <wei.zhi.liu@intel.com>

* install more binary files.

Signed-off-by: Liu, Wei Zhi <wei.zhi.liu@intel.com>

* update install files

Signed-off-by: Liu, Wei Zhi <wei.zhi.liu@intel.com>

---------

Signed-off-by: Liu, Wei Zhi <wei.zhi.liu@intel.com>
---
 debian/control |  2 +-
 debian/rules   | 11 ++++++++++-
 2 files changed, 11 insertions(+), 2 deletions(-)

diff --git a/debian/control b/debian/control
index 8b8fc26..43f8a47 100644
--- a/debian/control
+++ b/debian/control
@@ -8,7 +8,7 @@ Standards-Version: 4.1.4
 Package: orb-slam3
 Architecture: any
 Multi-Arch: foreign
-Depends: ${misc:Depends}, ${shlibs:Depends}, libopencv4.4, libeigen3.1.0, libboost-all-dev, libssl-dev, librealsense2-dev
+Depends: ${misc:Depends}, ${shlibs:Depends}, libopencv-dev, libeigen3-dev, libboost-all-dev, libssl-dev, librealsense2-dev
 Description: ORB-SLAM3: A robust and efficient SLAM system
  ORB-SLAM3 is a state-of-the-art Visual SLAM system for monocular, stereo, and RGB-D cameras, along with inertial 
  measurement units (IMU). It builds and maintains a 3D map of the environment using various sensor inputs and performs 
diff --git a/debian/rules b/debian/rules
index 55d7210..5a1e417 100755
--- a/debian/rules
+++ b/debian/rules
@@ -22,11 +22,20 @@ override_dh_install:
 # Install libraries and headers
 	install -d debian/orb-slam3/usr/lib/$(DEB_HOST_MULTIARCH)
 	install -d debian/orb-slam3/usr/include/orb-slam3
+	install -d debian/orb-slam3/opt/intel/orb-slam3/Vocabulary
+	install -d debian/orb-slam3/opt/intel/orb-slam3/Examples
 	ls -lah
 	install -m 0755 lib/libORB_SLAM3.so debian/orb-slam3/usr/lib/$(DEB_HOST_MULTIARCH)
+	install -m 0755 Thirdparty/g2o/lib/libg2o.so debian/orb-slam3/usr/lib/$(DEB_HOST_MULTIARCH)
+	install -m 0755 Thirdparty/DBoW2/lib/libDBoW2.so debian/orb-slam3/usr/lib/$(DEB_HOST_MULTIARCH)
 	cp -r include/* debian/orb-slam3/usr/include/orb-slam3
 
+	cp Vocabulary/ORBvoc.txt debian/orb-slam3/opt/intel/orb-slam3/Vocabulary
+	cp -r Examples/* debian/orb-slam3/opt/intel/orb-slam3/Examples
+	find debian/orb-slam3/opt/intel/orb-slam3 -name "*.cc" | xargs rm
+
+
 # Clean up any temporary files that were generated during the build
 override_dh_clean:
 # Perform cleanup after the build
-	dh_clean
\ No newline at end of file
+	dh_clean
-- 
2.30.2

